addons/website_sale/static/src/scss/website_sale.scss

```diff
@@ -405,15 +405,12 @@ a.no-decoration {
     }
 
     .carousel-outer {
-        height: 400px;
         max-height: 90vh;
         display: flex;
         align-items: center;
-        @include media-breakpoint-down(md) {
-            height: 300px;
-        }
-        @include media-breakpoint-down(sm) {
-            height: 200px;
+
+        &.o_carousel_not_single {
+            height: 25rem;
```

Comment:
This is unfortunately "not enough" 😕

| |
|--------|
|![image](https://github.com/user-attachments/assets/9abbeecf-58b5-4cf4-87fd-bc8c8f537e83)|

It's responsibility of the user to provide image of the "right size", yes,  but we should allow the use of portrait images here.
I propose:
```scss
&.o_carousel_not_single {
    height: 25rem;
  
    .product_detail_img {
        width: auto !important; // overrides .w-100
        max-height: 100%;
    }
}    
```

| |
|--------|
| ![image](https://github.com/user-attachments/assets/5a1ce709-1356-4129-a87d-37772a4c61b6) | 


...even better, we could keep the behavior described before while allowing the user to set a specific aspect-ratio  : 
```scss
 &.o_carousel_not_single {
            &.o_ratio-1x1 {
                aspect-ratio: 1 / 1;
            }

            &.o_ratio-4x3 {
                aspect-ratio: 4 / 3;
            }

            &.o_ratio-16x9 {
                aspect-ratio: 16 / 9;
            }

            &.o_ratio-21x9 {
                aspect-ratio: 21 / 9;
            }

            &.o_ratio-4x5 {
                aspect-ratio: 4 / 5;
            }
     
            .product_detail_img {
                width: auto !important; // overrides .w-100
                max-height: 100%;
            }
        }
  
    .product_detail_img {
        width: auto !important;
    }
}    
```

`addons/website_sale/views/snippets/snippets.xml`
```xml
<we-select string="Layout" data-name="o_wsale_image_layout" data-no-preview="true" data-reload="/">
    <we-button data-set-image-layout="carousel" data-name="o_wsale_carousel_opt">Carousel</we-button>
    <we-button data-set-image-layout="grid">Grid</we-button>
</we-select>
<we-select string="Images Ratio" data-name="o_wsale_carousel_height" data-apply-to=".o_carousel_product_outer" data-dependencies="o_wsale_carousel_opt" class="o_we_sublevel_1">
    <we-button data-select-class="o_ratio-4x5">Portait (4/5)</we-button>
    <we-button data-select-class="o_ratio-1x1">Square (1/1)</we-button>
    <we-button data-select-class="o_ratio-4x3">Landscape (4/3)</we-button>
    <we-button data-select-class="o_ratio-16x9">Wide (16/9)</we-button>
    <we-button data-select-class="o_ratio-21x9">Wider (21/9)</we-button>
</we-select>
```

|  |
|--------|
| ![image](https://github.com/user-attachments/assets/5702172f-8755-401d-ae9a-3caa32a139b1) |